coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -128,39 +131,47 @@ class QueryEngine(dataset: Dataset,
   private def walkLogicalPlanTree(logicalPlan: LogicalPlan,
                                   queryId: String,
                                   submitTime: Long,
-                                  options: QueryOptions): PlanResult = {
+                                  options: QueryOptions, spreadProvider: SpreadProvider): PlanResult = {
     logicalPlan match {
-      case lp: RawSeries =>                   materializeRawSeries(queryId, submitTime, options, lp)
-      case lp: RawChunkMeta =>                materializeRawChunkMeta(queryId, submitTime, options, lp)
-      case lp: PeriodicSeries =>              materializePeriodicSeries(queryId, submitTime, options, lp)
-      case lp: PeriodicSeriesWithWindowing => materializePeriodicSeriesWithWindowing(queryId, submitTime, options, lp)
-      case lp: ApplyInstantFunction =>        materializeApplyInstantFunction(queryId, submitTime, options, lp)
-      case lp: Aggregate =>                   materializeAggregate(queryId, submitTime, options, lp)
-      case lp: BinaryJoin =>                  materializeBinaryJoin(queryId, submitTime, options, lp)
-      case lp: ScalarVectorBinaryOperation => materializeScalarVectorBinOp(queryId, submitTime, options, lp)
-      case lp: LabelValues =>                 materializeLabelValues(queryId, submitTime, options, lp)
-      case lp: SeriesKeysByFilters =>         materializeSeriesKeysByFilters(queryId, submitTime, options, lp)
-      case lp: ApplyMiscellaneousFunction =>  materializeApplyMiscellaneousFunction(queryId, submitTime, options, lp)
+      case lp: RawSeries => materializeRawSeries(queryId, submitTime, options, lp, spreadProvider)
+      case lp: RawChunkMeta => materializeRawChunkMeta(queryId, submitTime, options, lp,
```

Comment:
This is MUCH less readable.  Please change back to previous alignment with `=>` lined up